app/src/main/java/com/android/sample/model/recipe/Recipe.kt

```diff
@@ -10,6 +21,9 @@ package com.android.sample.model.recipe
  * @property strInstructions Instructions on how to prepare the meal.
  * @property strMealThumbUrl URL of the thumbnail image of the meal.
  * @property ingredientsAndMeasurements A list of ingredient and measurement pairs for the recipe.
+ * @property time The time required to prepare the meal. Nullable.
+ * @property difficulty The difficulty level of the meal. Nullable.
+ * @property price The price of the meal. Nullable.
  */
 data class Recipe(
     val idMeal: String,
```

Comment:
Great job updating the `Recipe` data class and its `toFirestoreMap` method! 

**Quick Suggestions**:
1. **Default Values in Data Class**: For consistency, you could add default values to all nullable properties (like `strArea` and `strCategory`), making it easier to initialize without specifying every parameter.

   Example:
   ```kotlin
   val strArea: String? = null,
   ```

2. **Error Handling in Mapping**: Consider a check in `toFirestoreMap` for empty `ingredientsAndMeasurements` to prevent potential null-related errors in Firestore, especially if it’s required on the Firestore side.

Overall, this is looking neat and flexible—great work!

Reply:
Done!